tools/ts/build_lib.ts

```diff
@@ -0,0 +1,253 @@
+import { join } from "path";
```

Comment:
This file seems to be a top-level main script that depends on cwd. Because it's a lot of code, it would be good to put it inside a function with arguments as all the various inputs (e.g. cwd). This will potentially let us test it independently later.

Reply:
I was going to move it to args passed from the build script, because depending on other environments it simply wouldn't work (like when the build path is outside of the root of deno like on travis) but I was just "hacking" it to work until I was ready to integrate it, which is what I am doing now.